Explains how to import ts files instead of js files (#1056)
PrestaShop/docs: src/content/1.7/modules/concepts/templating/import-js.md

---
title: How to import Core JavaScript files 
menuTitle: Import Core JavaScript files
weight: 4
---

# How to import Core JavaScript files
{{< minver v="1.7.8" title="true">}}

> Most of the time, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may want to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.

Reviewer: ```suggestion
Most of the time, you will want to use the [global JavaScript components][global JavaScript components] object. However, in some cases, you may want to import a component using a static path, such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```

> **We do not recommend to do this**, but in case you absolutely need it, be careful to import the file without any extensions:

Reviewer: ```suggestion
**We don't recommend doing this.** You can still do it if you absolutely need it, but in that case, be careful to import the file without any extensions:
```

```javascript
// Incorrect
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.js'
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.ts'

// Correct
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search'
```

> If you need to carry out the above import, it is likely you use a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the TypeScript loader to your project and a `tsconfig` file.

Reviewer: ```suggestion
If you need to carry out the above import, it is likely you are using a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the TypeScript loader to your project and a `tsconfig` file.
```

---

> title: How to import core js files 

Reviewer 2: ```suggestion
title: How to import Core JavaScript files 
```

> menuTitle: Import core js files

Reviewer 2: Maybe a bit more formal? 😉 

```suggestion
menuTitle: Import Core JavaScript files
```

Reviewer: Maybe it's too long for the menu?

Author: Looks fine tbh 

![image](https://user-images.githubusercontent.com/14963751/125924356-5ebdfd44-f502-492c-a907-1de571d5f90c.png)

weight: 4
aliases:

> - 1.7/modules/templating/

Reviewer: Where does this alias come from? I don't see any page there currently.

Author: src/content/1.7/modules/concepts/templating/admin-views.md 

Comes from this file, I removed it in import-js, do you want me to remove it from admin-views?

Reviewer: Yep, that's a mistake, existing pages shouldn't be aliased.

---

> # How to import core js files

Reviewer: ```suggestion
# How to import Core JavaScript files
```

{{< minver v="1.7.8" title="true">}}

> Most of times, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may wants to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.

Reviewer 3: ```suggestion
Most of the time, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may want to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```

> **We do not recommend to do this**, but in case you absolutly need it be careful to import the file without any extensions:

Reviewer 2: ```suggestion
**We do not recommend to do this**, but in case you absolutely need it, be careful to import the file without any extensions:
```

```javascript
// Incorrect
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.js'
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.ts'

// Correct
import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search'
```

> As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to you're project and a `tsconfig` file.

Reviewer 3: ```suggestion
As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to your project and a `tsconfig` file.
```

> As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to your project and a `tsconfig` file.

Reviewer 2: ```suggestion
If you need to carry out the above import, it is likely you use a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the `TypeScript loader` to your project and a `tsconfig` file.
```

Example of a TypeScript loader:
```javascript
{
  test: /\.ts?$/,
  use: 'ts-loader',
  exclude: /node_modules/,
},
```

> Example of a TypeScript config file:

Reviewer 2: ```suggestion
Example of a TypeScript configuration file:
```
